unittests/APITests/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+set (TESTS
+  ReentrantMutex)
+
+list(APPEND LIBS FEXCore)
+
+foreach(API_TEST ${TESTS})
+  add_executable(${API_TEST} ${API_TEST}.cpp)
+  target_link_libraries(${API_TEST} PRIVATE ${LIBS} Catch2::Catch2WithMain)
+
+  catch_discover_tests(${API_TEST}
+    TEST_SUFFIX ".APITest")
+endforeach()
+
+execute_process(COMMAND "nproc" OUTPUT_VARIABLE CORES)
+string(STRIP ${CORES} CORES)
+
+add_custom_target(
+  api_tests
+  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
+  USES_TERMINAL
+  COMMAND "ctest" "--timeout" "302" "-j${CORES}" "-R" "\.*.APITest")
```

Comment:
Shouldn't this depend on the test targets? I've had weird build issues locally where CTest didn't recognize the test names until I had done a full build.
```cmake
foreach(API_TEST ${TESTS})
  add_dependency(api_tests ${API_TEST})
endforeach()
```

Reply:
Added dependencies.